beetsplug/unimported.py

```diff
@@ -40,11 +40,19 @@ def commands(self):
         def print_unimported(lib, opts, args):
             ignore_exts = [('.' + x).encode() for x
                            in self.config['ignore_extensions'].as_str_seq()]
-            in_folder = {
-                os.path.join(r, file) for r, d, f in os.walk(lib.directory)
-                for file in f if not any(
-                    [file.endswith(extension) for extension in
-                     ignore_exts])}
+            ignore_dirs = [lib.directory + ('/' + x.strip('/')).encode() for x
```

Comment:
Using `os.path.join` would be a bit safer (and easier to read) than the manual string manipulation. Similarly, `os.path.normpath` can do the stripping of `/` as well as other normalization that might be desirable in this situation.

Reply:
agreed - changed in the PR